configure.py

```diff
@@ -1420,6 +1424,106 @@ def bakefile_test_sources(file, sources):
 
     f.close()
 
+def gen_cmake(build_config, using_mods, cc, options):
+    def create_basic_rules(sources):
+        module = {'sources': {}, 'frameworks': set(), 'libs': set()}
+        for source in sources:
+            module['sources'][source] = {'isa_flags': set()}
+        return module
+
+    def update_rules(rules, using_mod):
+        needed = False
+        for source_path in rules['sources']:
+            for mod_source in using_mod.source:
+                if source_path == mod_source :
+                    needed = True
+                    for isa in using_mod.need_isa:
+                        rules['sources'][source_path]['isa_flags'].add(cc.isa_flags[isa])
+        if needed == True:
+            if options.os in using_mod.libs:
+                for lib in using_mod.libs[options.os]:
+                    rules['libs'].add(lib)
+            if options.os in using_mod.frameworks:
+                for framework in using_mod.frameworks[options.os]:
+                    rules['frameworks'].add(framework)
+
+    def generate_sources_list(fd, name, rules):
+        fd.write('set(' + name + '\n')
+        for source in rules['sources']:
+            fd.write('    ${CMAKE_CURRENT_LIST_DIR}/' + source[2:] + '\n')
+        fd.write(')\n\n')
+
+    def generate_special_properties(fd, rules):
+        for source in rules['sources']:
+            joined = ' '.join(rules['sources'][source]['isa_flags'])
+            if joined:
+                fd.write('set_source_files_properties(${CMAKE_CURRENT_LIST_DIR}/' + source[2:] + ' PROPERTIES COMPILE_FLAGS '+ joined + ')\n')
+
+    lib_module = create_basic_rules(build_config.sources)
+    test_module = create_basic_rules(build_config.test_sources)
+    cli_module = create_basic_rules(build_config.cli_sources)
```

Comment:
You are mixing the terms "rules" and "module" here which makes it very hard to follow the code flow even for an experienced cmake/Python coder. Isn't the proper term you are searching for "target" (which is what cmake uses to describe a static/shared library or an executable)?